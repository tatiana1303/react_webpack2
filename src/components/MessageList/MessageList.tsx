import React, { FC } from 'react';
import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';
import { Message } from '../../App';

interface MessageListProp {
  messages: Message[];
}

export const MessageList: FC<MessageListProp> = ({ messages }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        flexWrap: 'wrap',
        '& > :not(style)': {
          m: 1,
          width: 400,
          height: 400,
        },
      }}
    >
      <Paper>
        <ul>
          {messages.map((message) => (
            <li key={message.id} className="Message-text">
              {message.author}: {message.text}
            </li>
          ))}
        </ul>
      </Paper>
    </Box>
  );
};
