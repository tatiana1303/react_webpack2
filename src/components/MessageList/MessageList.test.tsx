import { MessageList } from './MessageList';

describe('MessageList', () => {
  it('to be Function', () => {
    expect(MessageList).toBeInstanceOf(Function);
  });

  it('should render MessageList', () => {
    expect(MessageList).toMatchSnapshot();
  });
});
