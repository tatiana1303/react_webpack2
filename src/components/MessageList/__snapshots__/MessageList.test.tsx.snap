// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render MessageList 1`] = `[Function]`;
